File: frontend/packages/shared/src/types/menu.ts
import type { Money } from "./auth";

export interface MenuItem {
  id: string;
  name: string;
  name_en: string;
  category_id: string;
  category_name: string;
  base_price: Money;
  channel_prices: ChannelPrice[];
  modifier_groups: ModifierGroup[];
  image_url: string | null;
  is_available: boolean;
}

export interface ChannelPrice {
  channel: "DineIn" | "Takeaway" | "Delivery";
  price: Money;
}

export interface MenuCategory {
  id: string;
  name: string;
  name_en: string;
  sort_order: number;
}

export interface ModifierGroup {
  id: string;
  name: string;
  name_en: string;
  selection_type: "single" | "multiple";
  is_required: boolean;
  options: ModifierOption[];
}

export interface ModifierOption {
  id: string;
  name: string;
  name_en: string;
  price: Money;
}

export interface Table {
  id: string;
  name: string;
  zone_id: string;
  zone_name: string;
  x: number;
  y: number;
  seats: number;
  status: "available" | "occupied" | "dirty" | "reserved";
  current_order_id: string | null;
}

export interface TableZone {
  id: string;
  name: string;
  floor: number;
}

export interface Promotion {
  id: string;
  name: string;
  name_en: string;
  trigger_type: string;
  condition: Record<string, unknown>;
  reward_type: string;
  reward: Record<string, unknown>;
  start_date: string;
  end_date: string;
  is_active: boolean;
}

File: backend/src/domain/entities/menu.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct MenuItem {
    pub id: Uuid,
    pub name: String,
    pub name_en: String,
    pub category_id: Uuid,
    pub category_name: Option<String>,
    pub base_price: i64,
    pub image_url: Option<String>,
    pub is_available: bool,
    pub deleted_at: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct MenuCategory {
    pub id: Uuid,
    pub name: String,
    pub name_en: String,
    pub sort_order: i32,
    pub item_count: Option<i64>,
    pub deleted_at: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct ModifierGroup {
    pub id: Uuid,
    pub menu_item_id: Uuid,
    pub name: String,
    pub name_en: String,
    pub selection_type: String,
    pub is_required: bool,
    pub sort_order: i32,
    pub deleted_at: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct ModifierOption {
    pub id: Uuid,
    pub modifier_group_id: Uuid,
    pub name: String,
    pub name_en: String,
    pub price: i64,
    pub sort_order: i32,
    pub deleted_at: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct ChannelPrice {
    pub id: Uuid,
    pub menu_item_id: Uuid,
    pub channel: String,
    pub price: i64,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
